File: Algorithms/560. Subarray Sum Equals K/golang/solution.go
func subarraySum(nums []int, k int) int {
	count := 0
	sum := 0
	m := map[int]int{
		0: 1,
	}
	for _, num := range nums {
		sum += num
		if val, ok := m[sum-k]; ok {
			count += val
		}
		m[sum]++
	}

	return count
}

File: Algorithms/560. Subarray Sum Equals K/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return Integer
     */
    function subarraySum($nums, $k)
    {
        $count = 0;
        $sum = 0;
        $map = [0 => 1];
        foreach ($nums as $num) {
            $sum += $num;
            if (isset($map[$sum - $k])) {
                $count += $map[$sum - $k];
            }
            $map[$sum] = isset($map[$sum]) ? $map[$sum] + 1 : 1;
        }
        return $count;
    }
}
